06.event_listeners/src/eventListeners.ts

```diff
@@ -0,0 +1,68 @@
+interface Data {
+  events: Map<string, EventListener[]>
+}
+
+const elementMap: Map<HTMLElement, Data> = new Map<HTMLElement, Data>();
+
+function eventHandler(event: Event) {
+  const { target } = event;
+  if (target instanceof HTMLElement && elementMap.has(target)) {
+    const callbackArray = elementMap.get(target)?.events.get(event.type);
+    callbackArray!.forEach((callback) => {
+      callback(event);
+    });
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const eventListener = {
+  on(element: HTMLElement, event: string, callback: EventListener) {
+    if (!elementMap.has(element)) {
+      elementMap.set(element, {
+        events: new Map([[event, [callback]]]),
+      });
+    } else {
+      const mapEvent = elementMap.get(element)!.events;
+      if (!mapEvent.has(event)) {
+        mapEvent.set(event, [callback]);
+      } else {
+        mapEvent.get(event)!.push(callback);
+      }
+    }
+    element.addEventListener(event, eventHandler);
+  },
+  off(element: HTMLElement, event?: string, callback?: EventListener) {
+    if (event) {
```

Comment:
you don't think it's prudent to removeEventListener after there is nothing there ?